### DISCUSSION TITLE
Image refinement after about 1 second

### DISCUSSION THREAD

**Discussion by DanielMGessel** [kovidgoyal/kitty#5847] [Q&A] [answered]
Happy kitty user but I'm seeing what may be the result of progressive display with the image protocol:  fine vertical and horizontal grid lines show up a little blotchy (bright/blurry in some areas), then clean up about one second later. I'm not sure if it's taking all that time to transfer the full image, or if kitty has an intentional delay.

If there's an intentional delay, can it be shortened (or eliminated, to display ASAP) with a parameter?

Probably irrelevant background:

I'm transferring as PNG, generated using cairosvg's svg2png. I can transfer the images via temp file or by sending the data directly in escape sequences. An obvious thing to try would be to save in non-progressive mode, but I haven't found an option for that

Investigating this, I spent a few minutes reading about PNG and it's "Adam7" progressive mode. It sees to reorder pixel transmission, so unlike jpeg, the early data is undersampled. The blotchiness is probably because my grid-lines are going in and out of phase with the grid nature of this undersampling.

**Comment by kovidgoyal** [comment, marked as answer]
There is no progressive rendering in kitty. All the image data is loaded
first, then rendered at once. See graphics.c

**Reply by DanielMGessel** [reply]
Thanks for the prompt reply! Leaves me with a mystery, but at least I know what it's not....

**Reply by DanielMGessel** [reply]
Just to follow up: the background is showing through when the image alpha is between 0 and 1 - strongest (I think) at 0.5. It's as if the destination is multiplied by 1-src_alpha then by src_alpha. The progressive effect seems to be the background fading out redraws in response to the blinking cursor. Using an opaque image solves the problem.

**Comment by kovidgoyal** [comment]
On Sun, Jan 01, 2023 at 07:08:50PM -0800, Daniel M Gessel wrote:
> Just to follow up: the background is showing through when the image alpha is between 0 and 1 - strongest (I think) at 0.5. It's as if the destination is multiplied by 1-src_alpha then by src_alpha. The progressive effect seems to be the background fading out redraws in response to the blinking cursor. Using an opaque image solves the problem.

You are likely on Wayland with a not up-to-date version of kitty.
